src/Asset.js

```diff
@@ -82,7 +82,11 @@ class Asset {
       .generateBundleName();
   }
 
-  mightHaveDependencies() {
+  async mightHaveDependencies() {
+    await this.loadIfNeeded();
+    if (!this.contents) {
```

Comment:
If `this.contents` is passed along a "falsy" value, such as an empty string, then `mightHaveDependencies` will return `false`, which might cause unintended behaviour.

Reply:
I am a little confused, this is indeed wants to skip `parse()` if `this.contents` is an empty string.